Adyen/UI/Form/Items/SelectableFormItem/SelectableFormItemView.swift

```diff
@@ -0,0 +1,180 @@
+        customButton.translatesAutoresizingMaskIntoConstraints = false
+        customButton.accessibilityIdentifier = item.identifier.map {
+            ViewIdentifierBuilder.build(scopeInstance: $0, postfix: "button")
+        }
+
+        customButton.addSubview(contentStackView)
+        contentStackView.isUserInteractionEnabled = false
+        contentStackView.adyen.anchor(inside: customButton)
+
+        return customButton
+    }()
+
+    @objc internal func didSelectItemButton() {
+        item.selectionHandler?()
+    }
+
+    /// Initializes the selectable form item view.
+    public required init(item: SelectableFormItem) {
+        super.init(item: item)
+        backgroundColor = item.style.backgroundColor
+
+        addSubview(itemButton)
+
+        updateItemData(item: item)
+        updateImageView(style: item.style)
+        configureConstraints()
+
+        preservesSuperviewLayoutMargins = true
+
+        itemButton.adyen.anchor(inside: self.layoutMarginsGuide)
+
+        observe(item.$isSelected) { [weak self] isSelected in
+            guard let self else { return }
+            self.checkmarkImageView.isHidden = !isSelected
+        }
+    }
+
+    override public func didMoveToWindow() {
+        super.didMoveToWindow()
+        updateIcon()
+    }
+
+    override public func layoutSubviews() {
+        super.layoutSubviews()
+
+        guard item.icon?.canBeModified == true else {
+            return imageView.adyen.round(using: .none)
+        }
+        imageView.adyen.round(using: item.style.imageStyle.cornerRounding)
+    }
+
+    private func updateImageView(style: SelectableFormItemStyle) {
+        imageView.contentMode = style.imageStyle.contentMode
+
+        guard item.icon?.canBeModified == true else {
+            return imageView.layer.borderWidth = 0
+        }
+
+        imageView.clipsToBounds = style.imageStyle.clipsToBounds
+        imageView.layer.borderWidth = style.imageStyle.borderWidth
+        imageView.layer.borderColor = style.imageStyle.borderColor?.cgColor
+    }
+
+    private func updateItemData(item: SelectableFormItem) {
+        accessibilityIdentifier = item.identifier
+
+        accessibilityLabel = item.accessibilityLabel
+        isAccessibilityElement = true
+
+        titleLabel.text = item.title
+        titleLabel.accessibilityIdentifier = item.identifier.map { ViewIdentifierBuilder.build(scopeInstance: $0, postfix: "titleLabel") }
+
+        imageView.isHidden = item.icon == nil
+        checkmarkImageView.isHidden = !item.isSelected
+        updateIcon()
+    }
+
+    private func updateIcon() {
+        if let iconUrl = item.icon?.url, window != nil, item.title != "Enter UPI ID" {
+            imageLoadingTask = imageView.load(url: iconUrl, using: imageLoader)
+        } else {
+            imageLoadingTask = nil
+        }
+    }
+
+    // MARK: - Layout
+
+    private let iconImageSize = CGSize(width: 40, height: 26)
```

Comment:
can be moved into the constants enum as well

Reply:
8079ceeba101a6e70258f18631cbfa9367c7b89e